descartes_light/include/descartes_light/interface/edge_evaluator.h

```diff
@@ -32,13 +32,59 @@ template <typename FloatType>
 class EdgeEvaluator
 {
 public:
+  using Ptr = typename std::shared_ptr<EdgeEvaluator<FloatType>>;
+  EdgeEvaluator(std::size_t dof) : dof_(dof) {}
   virtual ~EdgeEvaluator() {}
 
   virtual bool evaluate(const Rung_<FloatType>& from,
                         const Rung_<FloatType>& to,
-                        std::vector<typename LadderGraph<FloatType>::EdgeList>& edges) = 0;
+                        std::vector<typename LadderGraph<FloatType>::EdgeList>& edges)
```

Comment:
`edges` here is confusing. If I understand this correctly, this function returns a list of valid edges for each vertex (i.e. joint pose) in the `from` rung. This should probably be called something like `edge_lists` instead